import/src/test/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreServiceTest.java

```diff
@@ -128,107 +114,93 @@ public void setUp() throws Exception {
   }
 
   @Test
-  public void testCStoreService_basicCStore() throws Exception {
-    InputStream in =
-        new DicomInputStream(TestUtils.streamDICOMStripHeaders(TestUtils.TEST_MR_FILE));
-    InputStreamDataWriter data = new InputStreamDataWriter(in);
-
-    // Properties of the DICOM image.
-    String sopClassUID = UID.MRImageStorage;
-    String sopInstanceUID = "1.0.0.0";
-
-    // Create C-STORE DICOM server.
-    int serverPort = createDicomServer(/*connect error*/ false, HttpStatusCodes.STATUS_CODE_OK);
-
-    // Associate with peer AE.
-    Association association =
-        associate(serverHostname, serverPort, sopClassUID, UID.ExplicitVRLittleEndian);
-
-    // Send the DICOM file.
-    int wantStatus = Status.Success;
-    association.cstore(
-        sopClassUID,
-        sopInstanceUID,
-        1,
-        data,
-        UID.ExplicitVRLittleEndian,
-        createDimseRSPHandler(association, wantStatus));
-    association.waitForOutstandingRSP();
-
-    // Close the association.
-    association.release();
-    association.waitForSocketClose();
+  public void testCStoreService_success() throws Exception {
+    basicCStoreServiceTest(
+        false,
+        HttpStatusCodes.STATUS_CODE_OK,
+        Status.Success);
   }
 
   @Test
   public void testCStoreService_connectionError() throws Exception {
-    InputStream in =
-        new DicomInputStream(TestUtils.streamDICOMStripHeaders(TestUtils.TEST_MR_FILE));
-    InputStreamDataWriter data = new InputStreamDataWriter(in);
-
-    // Properties of the DICOM image.
-    String sopClassUID = UID.MRImageStorage;
-    String sopInstanceUID = "1.0.0.0";
+    basicCStoreServiceTest(
+        true,
+        HttpStatusCodes.STATUS_CODE_OK, // won't be returned due to connectionError
+        Status.ProcessingFailure);
+  }
 
-    // Create C-STORE DICOM server.
-    // The server is unable to get the response for the STOW-RS post (e.g. connection error).
-    int serverPort =
-        createDicomServer(/*connect error*/ true, HttpStatusCodes.STATUS_CODE_UNAUTHORIZED);
+  @Test
+  public void testCStoreService_unauthorized() throws Exception {
+    basicCStoreServiceTest(
+        false,
+        HttpStatusCodes.STATUS_CODE_UNAUTHORIZED,
+        Status.ProcessingFailure);
```

Comment:
Makes sense. I guess, I missed it because it wasn't specific to c-store. Ok, another pull request later (actually there may be more cases of general status codes like this).